src/wp-includes/global-styles-and-settings.php

```diff
@@ -407,21 +407,20 @@ function wp_theme_has_theme_json() {
 		return $theme_has_support;
 	}
 
+	$stylesheet_directory = get_stylesheet_directory();
+	$template_directory   = get_template_directory();
+
 	// This is the same as get_theme_file_path(), which isn't available in load-styles.php context
-	if ( file_exists( get_stylesheet_directory() . '/theme.json' ) ) {
-		$path = get_stylesheet_directory() . '/theme.json';
+	if ( $stylesheet_directory !== $template_directory && file_exists( $stylesheet_directory . '/theme.json' ) ) {
```

Comment:
@felixarntz @joemcgill This is a performance improvement, as on normal theme, stylesheet_directory and template_directory are the same.